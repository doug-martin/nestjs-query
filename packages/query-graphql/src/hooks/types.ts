export enum HookTypes {
  BEFORE_CREATE_ONE = 'BeforeCreateOne',
  BEFORE_CREATE_MANY = 'BeforeCreateMany',
  BEFORE_UPDATE_ONE = 'BeforeUpdateOne',
  BEFORE_UPDATE_MANY = 'BeforeUpdateMany',
  BEFORE_DELETE_ONE = 'BeforeDeleteOne',
  BEFORE_DELETE_MANY = 'BeforeDeleteMany',
  BEFORE_QUERY_MANY = 'BeforeQueryMany',
  BEFORE_FIND_ONE = 'BeforeFindOne',
}
