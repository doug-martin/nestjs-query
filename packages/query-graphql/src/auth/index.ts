export * from './authorizer';
export * from './default-crud.authorizer';
export * from './tokens';
