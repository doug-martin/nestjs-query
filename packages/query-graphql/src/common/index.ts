export { DTONamesOpts, getDTONames, DTONames, getDTOIdTypeOrDefault } from './dto.utils';
export * from './external.utils';
export * from './resolver.utils';
export * from './object.utils';
