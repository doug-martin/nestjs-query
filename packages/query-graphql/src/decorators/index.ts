export {
  FilterableField,
  FilterableFieldOptions,
  FilterableFieldDescriptor,
  getFilterableFields,
} from './filterable-field.decorator';
export { ResolverMethodOpts } from './resolver-method.decorator';
export {
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  Relation,
  FilterableRelation,
  RelationDecoratorOpts,
  RelationTypeFunc,
  getRelations,
} from './relation.decorator';
export * from './resolver-mutation.decorator';
export * from './resolver-query.decorator';
export * from './resolver-field.decorator';
export { Reference, ReferenceDecoratorOpts, ReferenceTypeFunc } from './reference.decorator';
export { ResolverSubscription, SubscriptionResolverMethodOpts } from './resolver-subscription.decorator';
export { InjectPubSub } from './inject-pub-sub.decorator';
export * from './skip-if.decorator';
export * from './aggregate-query-param.decorator';
export * from './hook.decorator';
export * from './decorator.utils';
export * from './hook-args.decorator';
export * from './authorizer.decorator';
export * from './inject-authorizer.decorator';
export * from './inject-custom-authorizer.decorator';
export * from './key-set.decorator';
export * from './authorize-filter.decorator';
export * from './query-options.decorator';
export * from './id-field.decorator';
