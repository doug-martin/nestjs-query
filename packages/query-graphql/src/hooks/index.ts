export * from './hooks';
export { createDefaultHook } from './default.hook';
export * from './tokens';
export * from './types';
