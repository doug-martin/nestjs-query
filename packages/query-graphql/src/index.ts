export * from './types';
export {
  FilterableField,
  FilterableFieldOptions,
  ResolverMethodOpts,
  Relation,
  FilterableRelation,
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  RelationTypeFunc,
  RelationDecoratorOpts,
  Reference,
  ReferenceTypeFunc,
  ReferenceDecoratorOpts,
  InjectPubSub,
  BeforeCreateOne,
  BeforeCreateMany,
  BeforeUpdateOne,
  BeforeUpdateMany,
  BeforeDeleteOne,
  BeforeDeleteMany,
  BeforeQueryMany,
  BeforeFindOne,
  InjectAuthorizer,
  InjectCustomAuthorizer,
  Authorize,
  AuthorizerFilter,
  RelationAuthorizerFilter,
  HookArgs,
  MutationHookArgs,
  KeySet,
  QueryOptions,
  IDField,
  IDFieldOptions,
} from './decorators';
export * from './resolvers';
export * from './federation';
export { DTONamesOpts } from './common';
export { NestjsQueryGraphQLModule } from './module';
export { AutoResolverOpts } from './providers';
export { pubSubToken, GraphQLPubSub } from './subscription';
export { Authorizer, CustomAuthorizer, AuthorizerOptions, AuthorizationContext, OperationGroup } from './auth';
export {
  Hook,
  HookTypes,
  BeforeCreateOneHook,
  BeforeCreateManyHook,
  BeforeUpdateOneHook,
  BeforeUpdateManyHook,
  BeforeDeleteOneHook,
  BeforeDeleteManyHook,
  BeforeQueryManyHook,
  BeforeFindOneHook,
} from './hooks';
export { AuthorizerInterceptor, AuthorizerContext, HookInterceptor, HookContext } from './interceptors';
