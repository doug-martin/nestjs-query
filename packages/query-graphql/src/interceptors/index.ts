export * from './hook.interceptor';
export * from './authorizer.interceptor';
