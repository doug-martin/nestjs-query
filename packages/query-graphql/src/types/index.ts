export {
  PageInfoType,
  EdgeType,
  OffsetPageInfoType,
  ConnectionType,
  ConnectionOptions,
  CursorConnectionType,
  CursorConnectionOptions,
  ArrayConnectionType,
  ArrayConnectionOptions,
  OffsetConnectionType,
  OffsetConnectionOptions,
  StaticConnectionType,
  InferConnectionTypeFromStrategy,
} from './connection';
export { MutationArgsType } from './mutation-args.type';
export {
  FilterType,
  UpdateFilterType,
  DeleteFilterType,
  SubscriptionFilterType,
  QueryArgsType,
  QueryArgsTypeOpts,
  PagingTypes,
  CursorQueryArgsType,
  OffsetQueryArgsType,
  NonePagingQueryArgsType,
  CursorPagingType,
  OffsetPagingType,
  PagingStrategies,
} from './query';
export { CreateManyInputType } from './create-many-input.type';
export { CreateOneInputType } from './create-one-input.type';
export { UpdateManyResponseType } from './update-many-response.type';
export { UpdateOneInputType } from './update-one-input.type';
export { UpdateManyInputType } from './update-many-input.type';
export { DeleteManyResponseType } from './delete-many-reponse.type';
export { DeleteOneInputType } from './delete-one-input.type';
export { DeleteManyInputType } from './delete-many-input.type';
export { ConnectionCursorType, ConnectionCursorScalar } from './cursor.scalar';
export { RelationsInputType } from './relations-input.type';
export { RelationInputType } from './relation-input.type';
export { SubscriptionArgsType } from './subscription-args.type';
export { SubscriptionFilterInputType } from './subscription-filter-input.type';
export { AggregateResponseType, AggregateResponseOpts, AggregateArgsType } from './aggregate';
export { FindOneArgsType } from './find-one-args.type';
