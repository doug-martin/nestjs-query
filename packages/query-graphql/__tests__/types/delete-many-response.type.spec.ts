import { Resolver, Query } from '@nestjs/graphql';
import { DeleteManyResponse } from '@nestjs-query/core';
import { DeleteManyResponseType } from '../../src/types';
import { generateSchema } from '../__fixtures__';

describe('DeleteManyResponseType', (): void => {
  it('should create a @nestjs/graphql object type', async () => {
    const D = DeleteManyResponseType();
    @Resolver()
    class TestDeleteManyResponseResolver {
      @Query(() => D)
      deleteTest(): DeleteManyResponse {
        return { deletedCount: 1 };
      }
    }
    const schema = await generateSchema([TestDeleteManyResponseResolver]);
    expect(schema).toMatchSnapshot();
  });
});
