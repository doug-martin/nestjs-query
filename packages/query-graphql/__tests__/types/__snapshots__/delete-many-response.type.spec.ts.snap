// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteManyResponseType should create a @nestjs/graphql object type 1`] = `
type DeleteManyResponse {
  """The number of records deleted."""
  deletedCount: Int!
}

type Query {
  deleteTest: DeleteManyResponse!
}

`;
